spring-kafka/src/main/java/org/springframework/kafka/listener/adapter/MessagingMessageListenerAdapter.java

```diff
@@ -869,6 +886,17 @@ private boolean rawByParameterIsType(Type parameterType, Type type) {
 		return parameterType instanceof ParameterizedType pType && pType.getRawType().equals(type);
 	}
 
+	/**
+	 * Sets the retry callback for failures of both {@link CompletableFuture} and {@link Mono}.
```

Comment:
Method JavaDocs have to be imperative, like commands.
Therefore `Set the retry callback...`.
Also `@since 3.3` still has to be present on the public API.